graphqlmetrics/core/metrics_service.go

```diff
@@ -161,80 +161,59 @@ func (s *MetricsService) prepareClickhouseBatches(
 		operationBatch, metricBatch driver.Batch
 	)
 
-	processableOperationItems := make([]*graphqlmetricsv1.SchemaUsageInfo, 0)
+	metricsBatchPrepared := false
 
 	for _, item := range batch {
-		for _, schemaUsage := range item.SchemaUsage {
-			// Skip if there are no request document
-			if schemaUsage.RequestDocument == "" {
-				continue
+		for _, su := range item.SchemaUsage {
+
+			if !metricsBatchPrepared && (len(su.TypeFieldMetrics) > 0 || len(su.ArgumentMetrics) > 0 || len(su.InputMetrics) > 0) {
+				metricsBatchPrepared = true
+
+				// If any of the schema usage items has metrics to process, we need to ensure the metric batch is prepared once.
+				metricBatch, err = s.conn.PrepareBatch(ctx, `INSERT INTO gql_metrics_schema_usage`)
+				if err != nil {
+					return nil, nil, fmt.Errorf("failed to prepare metric batch for metrics: %w", err)
+				}
 			}
-			// If the operation is already in the cache, we can skip it and don't write it again
-			if _, exists := s.opGuardCache.Get(schemaUsage.OperationInfo.Hash); exists {
+			
+			err := s.appendUsageMetrics(metricBatch, insertTime, item.Claims, su)
```

Comment:
We don't want to abort operation insert when appending a metric errored. The way how we handled it previously was intentionally.

Reply:
Makes sense. I will log the error instead and include that in the test